### ISSUE TITLE
Players may start to desync / stutter in networked play

### ISSUE THREAD

**Issue by MaxCWhitehead** [fishfolk/jumpy#918]
### Description

I found that after jumping around / moving for a bit in a networked match, I started to see the remote player stutter, on tapping right, they'd move forward a ways then snap back to expected position.  It occurs in both directions (each client stutters movement on opposite).

~~I have local changes testing smoothing net corrections, but when I disable this it still occurs. I'll triple check it's not my local changes by stashing them though.~~
EDIT: confirmed not my local changes, repro'd on main.

Reproduced a few times.

### To Reproduce

open network match with two clients, move and jump around for a bit.

**Comment by MaxCWhitehead** [comment]
Here's a video

https://github.com/fishfolk/jumpy/assets/35712032/3c0e1649-e27b-4ae7-ae11-bd5894c45505

**Comment by MaxCWhitehead** [comment]
Current theory is that the stutter is due to simulation advancing a few frames with predicted input, and once a confirmed input is received indicating the button was no longer pressed, it corrects and rolls back to expected position.

So this would indicate unreliable network conditions or high latency. I should get the network debug tools back online. I should also really go to bed lmao, but I'll test with a local matchmaking server or LAN and see if that helps.

I tested some much older commits, don't think it's a recent bug? I've been in this setup all the time and just started hitting this tonight. I have a fear it'll go away come tomorrow. My laptop could also just be screwy. Either way would be good to understand what conditions cause this or figure out what the bug is if doing something wrong.

**Comment by MaxCWhitehead** [comment]
<img width="1339" alt="image" src="https://github.com/fishfolk/jumpy/assets/35712032/82864f96-8ff8-4e30-b557-16b9aec134b5">

It still reproduces when neither clients are at max predicted frames (not freezing), so can rule that hypothesis out. Will see if it correlates with FPS next probably and that we are not somehow stepping multiple times per network frame or something.

Got network debug menu working (and entirely in bones), so silver linings.

**Comment by MaxCWhitehead** [comment]
Testing with local docker client seemed to help, I feel like the problem might still be there just on a very small scale. Definitely seems to be exasperated by ping. My ping to match making server seems somewhat unstable, will look into that, not sure if it is the only cause.

**Comment by MaxCWhitehead** [comment]
So this isn't actually a bug, but an artifact of a double correction when input is tapped. Made worse by large frame difference.

Scenario:
- Remote player idles, then taps right.
- Client gets input 5 frames in past, rolls back. (remote player was standing still, rollback to same position).
  - Client repredicts 5 frames, predicting that remote player held right down entire time.
  - Frame is rendered, there is a pop of 5 frames of movement to right.
- Remote player's 2nd input of releasing tap is received
- Client rolls back again, repredicts, and snaps back to slightly past where it started.

Will think about options here, there are probably gonna be some level of artifacts at 200 ping, maybe this will get better with some of the net correction smoothing stuff I was doing. Think I've gone full circle.

I don't know why I just started hitting this, either my ping got worse, or I started tapping more than I normally do while testing.
